aspnetcore/fundamentals/routing.md

```diff
@@ -310,30 +310,37 @@ Route constraints execute when a `Route` has matched the syntax of the incoming
 
 The following table demonstrates some route constraints and their expected behavior.
 
-| constraint | Example | Example Match | Notes |
+| constraint | Example | Example Matches | Notes |
 | --------   | ------- | ------------- | ----- |
-| `int` | {id:int} | 123 | Matches any integer |
-| `bool`  | {active:bool} | true | Matches `true` or `false` |
-| `datetime` | {dob:datetime} | 2016-01-01  | Matches a valid `DateTime` value (in the invariant culture - see options) |
-| `decimal` | {price:decimal} | 49.99 | Matches a valid `decimal` value |
-| `double`  | {weight:double} | 4.234 | Matches a valid `double` value |
-| `float`  | {weight:float} | 3.14 | Matches a valid `float` value |
-| `guid`  | {id:guid} | CD2C1638-1638-72D5-1638-DEADBEEF1638 | Matches a valid `Guid` value |
-| `long` | {ticks:long} | 123456789 | Matches a valid `long` value |
-| `minlength(value)` | {username:minlength(4)} | Rick | String must be at least 4 characters |
-| `maxlength(value)` | {filename:maxlength(8)} | somefile | String must be no more than 8 characters |
-| `length(min,max)` | {filename:length(8,16)} | *Somefile.txt* | String must be at least 8 and no more than 16 characters long |
-| `min(value)` | {age:min(18)} | 19 | Value must be at least 18 |
-| `max(value)` | {age:max(120)} |  91 | Value must be no more than 120 |
-| `range(min,max)` | {age:range(18,120)} | 91 | Value must be at least 18 but no more than 120 |
-| `alpha` | {name:alpha} | Rick | String must consist of alphabetical characters |
-| `regex(expression)` | {ssn:regex(^d{3}-d{2}-d{4}$)} | 123-45-6789 | String must match the regular expression |
-| `required`  | {name:required} | Rick |  Used to enforce that a non-parameter value is present during URL generation |
+| `int` | `{id:int}` | `123456789`, `-123456789`  | Matches any integer |
```

Comment:
I added more esoteric examples here to give much more info on what kind of values could be parsed (e.g. whether thousands separators, exponents, negative values, case-sensitivity, etc. were supported). There were also a few missing built-in constraints, such as `length(length)`.